src/controllers/lessons.js

```diff
@@ -8,196 +8,226 @@ import Homework from "../models/Homework.js";
-			motivationTitle: req.body.motivationTitle,
-			cohort: req.body.cohort,
-			note: req.body.note,
-			timestamps: timestamps,
-		};
-		const lesson = await Lesson.findByIdAndUpdate(
-			req.params.id || mongoose.Types.ObjectId(),
-			lessonData,
-			{ upsert: true, new: true }
-		);
-
-		// if this is a new class, update all users with current class = null so this is now their current class
-		if (!req.params.id) {
-			await User.updateMany(
-				{ currentClass: null },
-				{ currentClass: lesson._id }
-			);
-		}
-		req.session.flash = {
-			type: "success",
-			message: [`Class ${!!req.params.id ? "updated" : "added"}`],
-		};
-	} catch (err) {
-		console.log(err);
-		req.session.flash = {
-			type: "error",
-			message: [`Class not ${!!req.params.id ? "updated" : "added"}`],
-		};
-	} finally {
-		res.redirect("/class/add");
-	}
+  if (!req.isAuthenticated() || !req.user.admin) return res.redirect("/");
+  try {
+    let dates = [];
+    if (req.body.date) {
+      dates = req.body.date.split(",").map((date) => new Date(date));
+    }
+    let slides = [];
+    const timestamps = [];
+    for (let i = 0; i < req.body.tsTime.length; i++) {
+      timestamps.push({
+        time: Number(req.body.tsTime[i]),
+        title: req.body.tsTitle[i],
+      });
+    }
+    const lessonData = {
+      videoId: req.body.videoId,
+      twitchVideo: !!req.body.twitch ? true : false,
+      title: req.body.videoTitle,
+      dates: dates,
+      permalink: req.body.permalink,
+      thumbnail: req.body.thumbnail,
+      classNo: req.body.number ? req.body.number.split(",") : [],
+      slides: req.body.slides ? req.body.slides.split(",") : [],
+      materials: req.body.materials,
+      checkin: req.body.checkin ? req.body.checkin.split(",") : [],
+      motivationLink: req.body.motivationLink,
+      motivationTitle: req.body.motivationTitle,
+      cohort: req.body.cohort,
+      note: req.body.note,
+      tags: req.body.tags
+        ? req.body.tags
+            .split(",")
+            .map((tag) => tag.trim().toLowerCase())
+            // add space after comma
+            .map((tag) => tag.replace(/,/g, ", "))
+        : [],
+      timestamps: timestamps,
+    };
+    const lesson = await Lesson.findByIdAndUpdate(
+      req.params.id || mongoose.Types.ObjectId(),
+      lessonData,
+      { upsert: true, new: true }
+    );
+
+    // if this is a new class, update all users with current class = null so this is now their current class
+    if (!req.params.id) {
+      await User.updateMany(
+        { currentClass: null },
+        { currentClass: lesson._id }
+      );
+    }
+    req.session.flash = {
+      type: "success",
+      message: [`Class ${!!req.params.id ? "updated" : "added"}`],
+    };
+  } catch (err) {
+    console.log(err);
+    req.session.flash = {
+      type: "error",
+      message: [`Class not ${!!req.params.id ? "updated" : "added"}`],
+    };
+  } finally {
+    res.redirect(`/class/edit/${req.params.id}`);
```

Comment:
When submitting an "edit class" form, it redirected to a new blank form, this redirects back to the same class